src/Microsoft.SqlTools.SqlCore/Scripting/ScriptAsScriptingOperation.cs

```diff
@@ -53,6 +53,12 @@ public ScriptAsScriptingOperation(ScriptingParams parameters, string azureAccoun
             }
 
             ServerConnection = new ServerConnection(sqlConnection);
+            if (string.IsNullOrEmpty(ServerConnection.Password)) {
+                // Manually add password as server connection class does not add it to string, required for successful connection.
+                if(ServerConnection.ConnectionString.IndexOf("Authentication=SqlPassword") != -1 && ServerConnection.ConnectionString.IndexOf("Password=") == -1) {
```

Comment:
Instead of comparing strings, you can instantiate Sql Connection string builder with this connection string and then compare options.